rust/lance-file/src/page_table.rs

```diff
@@ -99,10 +127,10 @@ impl PageTable {
             .unwrap()
             + 1;
 
-        let mut builder = Int64Builder::with_capacity((num_columns * num_batches) as usize);
-        for col in 0..num_columns {
+        let mut builder =
+            Int64Builder::with_capacity(field_ids.clone().count() * num_batches as usize);
```

Comment:
Minor nit but I think you can avoid the clone with `field_ids.len()`

Reply:
`ExactSizeIterator` isn't implemented for `i32`, unfortunately. I think because it's possible to construct a range that won't fit into 32-bit `usize`.